hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSetProcessor.java

```diff
@@ -412,11 +412,14 @@ public StructType getExternalFileSchema(String storedAs, String location) throws
     @Override
     public void createEmptyExternalFile(ExecRow execRows, int[] baseColumnMap, int[] partitionBy, String storedAs,  String location, String compression) throws StandardException {
         try{
-
             StructType schema = AvroUtils.supportAvroDateType(execRows.schema(),storedAs);
-
-            Dataset<Row> empty = SpliceSpark.getSession()
-                        .createDataFrame(new ArrayList<Row>(), schema);
+            SparkSession spliceSpark = SpliceSpark.getSession();
+            if (storedAs != null && storedAs.toLowerCase().equals("a")){
+                if (compression.toLowerCase().equals("zlib"))
+                    compression = "deflate";
+                spliceSpark.conf().set("spark.sql.avro.compression.codec",compression);
+            }
```

Comment:
spark session and configuration is shared by all spark jobs. If two "create external table" are being executed, and one wants to compress avro file, the other does not, or they try to compress in different format, setting a global configuration may not work

Reply:
Thanks @jyuanca  Global session is improper, let me address it.